tests/test_csas.py

```diff
@@ -0,0 +1,163 @@
+        ('SBSP', datetime(2003, 1, 1),
+         datetime(2003, 1, 1), ['SBSP_1hr_2003-2009.csv']),
+        ('SBSP', datetime(2010, 1, 1),
+         datetime(2010, 1, 1), ['SBSP_1hr_2010-2023.csv']),
+        # Test straddling the files
+        ('SBSP', datetime(2008, 1, 1),
+         datetime(2010, 1, 1), ['SBSP_1hr_2003-2009.csv', 'SBSP_1hr_2010-2023.csv']),
+        # test stations that don't have year specific timeframes
+        ('PTSP', datetime(2010, 1, 1), datetime(2010, 1, 2), ["PTSP_1hr.csv"]),
+        ('SBSG', datetime(2010, 1, 1), datetime(2010, 1, 2), ["SBSG_1hr.csv"])
+    ])
+    def test_file_urls(self, station_id, start, end, expected):
+        pnt = CSASMet(station_id)
+        pnt._verify_station()
+        urls = pnt._file_urls(station_id, start, end)
+        names = sorted([Path(url).name for url in urls])
+        assert names == expected
+
+    @pytest.mark.parametrize("station_id, start, end", [
+        # Test pre 2003
+        ('SBSP', datetime(2002, 1, 1),
+         datetime(2003, 1, 1)),
+        # Test post 2023 exception
+        ('SBSP', datetime(2010, 1, 1),
+         datetime(2024, 1, 1))
+    ])
+    def test_file_urls_exception(self, station_id, start, end):
+        """
+        With files there are hard timeframes, test ane exception is raised when
+        this is the case
+        """
+        pnt = CSASMet(station_id)
+        pnt._verify_station()
+        with pytest.raises(InvalidDateRange):
+            pnt._file_urls(station_id, start, end)
+
+    @pytest.mark.parametrize("station_id, year", [
+        # Test the two SBSP urls
+        ('SBSP', 2009),
+        ('SBSP', 2010),
+        # Test the two SASP urls
+        ('SASP', 2009),
+        ('SASP', 2010),
+        # Test the Stream gauges url
+        ('SBSG', 2005),
+        # Test the PTSP url
+        ('PTSP', 2003),
+    ])
+    def test_links_are_valid(self, station_id, year):
+        """
+        Seeking answers from CSAS on how and when these files are updated. Until then
+        this will serve as a nice way to check the files are workin still
+        """
+        start = datetime(year, 1, 1)
+        end = start + timedelta(days=1)
+
+        pnt = CSASMet(station_id)
+        pnt._verify_station()
+        urls = pnt._file_urls(station_id, start, end)
+
+        resp = requests.head(urls[0])
+        assert resp.ok
+
+    @pytest.mark.parametrize('station_id, variable, start, end, expected_mean', [
+        ('SASP', CSASVariables.SURF_TEMP, DT_20230301, DT_20230315, -11.59220),
+        # Span two files, Test data wont be complete but requires two files
+        ('SASP', CSASVariables.SURF_TEMP, DT_20090301, DT_20230315, -10.35284),
+    ])
+    def test_get_daily_data(self, station, station_id, variable, start, end,
+                            expected_mean):
+        """ Check pulling two weeks of data """
+
+        df = station.get_daily_data(start, end, [variable])
```

Comment:
would it make sense to mock (or patch i guess) requests so we can check that it is not called since the files exist? You can do an .assert_not_called() on mock objects

Reply:
Yes...let me do that. At the time I was feeling the desire to move on so I did the easiest thing I could think of.